extensions/s3/src/main/java/io/deephaven/extensions/s3/S3SeekableChannelProvider.java

```diff
@@ -73,15 +74,28 @@ final class S3SeekableChannelProvider implements SeekableChannelsProvider {
 
     private volatile SoftReference<Map<URI, FileSizeInfo>> fileSizeCacheRef;
 
+    /**
+     * Used sanitizing URIs at the beginning of all public APIs before using the URI in S3 requests.
+     */
+    private final Function<URI, URI> uriSanitizer;
```

Comment:
This approach seems okay; the other approach we _could_ take is `GcsSeekableChannelProvider` that does the sanitization and then calls out to `S3SeekableChannelProvider`. It could also update the scheme on the way out. I wonder if that is better, and then we don't need to do the relativize change I think?